e107_plugins/pm/pm.php

```diff
@@ -373,17 +388,21 @@ function show_pm($pmid, $comeFrom = '')
 	public function showBlocked($start = 0)
 	{
 		$tp = e107::getParser();
-		$tpl_file = THEME.'pm_template.php';
-		include(is_readable($tpl_file) ? $tpl_file : e_PLUGIN.'pm/pm_template.php');
+//		$tpl_file = THEME.'pm_template.php';
+//		include(is_readable($tpl_file) ? $tpl_file : e_PLUGIN.'pm/pm_template.php');
+    $PM_BLOCKED = e107::getTemplate('pm', 'pm', 'blocked');
+
 		$pmBlocks = $this->block_get_user();			// TODO - handle pagination, maybe (is it likely to be necessary?)
 
 		$sc = e107::getScBatch('pm',TRUE, 'pm');
 		$sc->pmBlocks = $pmBlocks; 
+  $sc->wrapper('pm');
 	
-		$PM_BLOCKED_HEADER = $this->updateTemplate($PM_BLOCKED_HEADER);
-		$PM_BLOCKED_TABLE  = $this->updateTemplate($PM_BLOCKED_TABLE);
-		$PM_BLOCKED_EMPTY  = $this->updateTemplate($PM_BLOCKED_EMPTY);
-		$PM_BLOCKED_FOOTER  = $this->updateTemplate($PM_BLOCKED_FOOTER);
+// Is updateTemplate really necessary for v2.x templates?
+		$PM_BLOCKED_HEADER = $this->updateTemplate($PM_BLOCKED['start']);
+		$PM_BLOCKED_TABLE  = $this->updateTemplate($PM_BLOCKED['item']);
+		$PM_BLOCKED_EMPTY  = $this->updateTemplate($PM_BLOCKED['empty']);
+		$PM_BLOCKED_FOOTER  = $this->updateTemplate($PM_BLOCKED['end']);
 	
```

Comment:
I thought of that when doing the changes, but how can you deal with a V1 template when the plugin itself is upgraded to V2 standards?
V1.x theme templates were in the theme root, V2.x theme templates are on the templates/plugin folder, and therefore, if the plugin uses the V2.x gettemplate core function, it won't simply pick up the legacy V1.x template, and will pick up it's own plugin "default" template.
So, for me, as i see it, moving the old legacy template to the v2.x theme template location only will add confusion, as the template won't be a real V2.x template on a V2.x theme template location, so it won't work neither...
Theme designers should upgrade their templates to V2.x standards, as old V1.x custom plugins templates simply won't work on upgraded plugins to V2.x standards....
Do i see it wrong, or not?

We can always do a theme templates conversion on the fly function, to convert v1.x theme templates to V2.x standars, but it would be a waste of time, as i see it.
The use of V1.x templates should be discouraged, or at least, V1.x templates with custom plugins templates on it. It should be a visible warning for the admin, about V1.x plugins templates present in the current theme that the admin is trying to use, in order to prevent questions like: "why my v1.x theme custom template is not working?". The warning should also advert the admin that those theme legacy plugnin templates should be upgraded to V2.x standards...

P.S.: Sorry for the long text, i hope i explained my point of view...